tests/scalability/generate_brac_data/generate-brac-data.js

```diff
@@ -0,0 +1,217 @@
+const bracPlaceFactory = require('../../factories/brac/contacts/brac-place');
+const bracPersonFactory = require('../../factories/brac/contacts/brac-person');
+const bracUserFactory = require('../../factories/brac/users/brac-user');
+const bracSurvey = require('../../factories/brac/reports/brac-survey');
+const fs = require('fs');
+const Faker = require('@faker-js/faker');
+const dataConfig = require('./data-config.json');
+const sizeConfig = require('./size-config.json');
+const createCsvWriter = require('csv-writer').createObjectCsvWriter;
+const [, , threadId] = process.argv;
+const districtHospitalName = `${threadId}-districthospital`;
+const numberOfDistrictHospitals = sizeConfig.number_of_district_hospitals;
+const numberOfManagersPerDistrictHospital = sizeConfig.number_of_managers_per_district_hospitals;
+const numberOfHealthCentersPerDistrictHospital = sizeConfig.number_of_health_centers_per_district_hospital;
+const numberOfChwPerHealthCenter = sizeConfig.number_of_chw_per_health_center;
+const numberOfClinicsPerHealthCenter = sizeConfig.number_of_clinics_per_health_center;
+const numberOfFamilyMembers = sizeConfig.number_of_family_members;
+const dataDirectory = dataConfig.data_directory;
+const preconditionDirectory = dataConfig.precondition_data_directory;
+const mainDirectory = dataConfig.main_script_data_directory;
+const jsonDirectory = dataConfig.json_directory;
+const preconditionDataDirectory = dataConfig.data_directory
+  + dataConfig.precondition_data_directory + dataConfig.json_directory;
+const mainDataDirectory = dataConfig.data_directory + dataConfig.main_script_data_directory;
+const usersDirectory = dataConfig.data_directory + dataConfig.precondition_data_directory;
+const dataExtension = dataConfig.json_data_extension;
+const csvWriter = createCsvWriter({
+  path: usersDirectory + 'users.csv',
+  header: [
+    { id: 'username', title: 'username' },
+    { id: 'password', title: 'password' },
+    { id: 'roles', title: 'roles' },
+    { id: 'contact', title: 'contact' },
+    { id: 'phone', title: 'phone' },
+    { id: 'place', title: 'place' },
+  ],
+  alwaysQuote: true,
+  append: true
+});
+
+const users = [];
+const managers = [];
+
+const createDataDoc = (directory, content) => {
+  fs.writeFile(directory, content, err => {
+    if (err) {
+      console.error('CreateDataDoc ' + err);
+      return;
+    }
+  });
+};
+
+const createDataDirectory = (path, directoryName) => {
+  if (!fs.existsSync(path + directoryName)) {
```

Comment:
It's also best practice to not rely on `existsSync` before doing other operations on that path: 
> Using fs.exists() to check for the existence of a file before calling fs.open(), fs.readFile() or fs.writeFile() is not recommended. Doing so introduces a race condition, since other processes may change the file's state between the two calls. Instead, user code should open/read/write the file directly and handle the error raised if the file does not exist.

(source: https://nodejs.org/api/fs.html#fsexistspath-callback)

So it's best if this becomes something like: 
```javascript
const createDataDirectory = async (directoryPath, directoryName) => {
  try {
    await fs.promises.mkdir(path.join(directoryPath, directoryName));
  } catch (err) {
    if (err.code === 'EEXIST') {
      return;
    }
    console.error('CreateDataDirectory ' + directoryName + ' failed - ' + err);
    throw err;
  }
};
```